ext/ddtrace_profiling_native_extension/collectors_discrete_dynamic_sampler.c

```diff
@@ -28,6 +43,7 @@ static void _discrete_dynamic_sampler_reset(discrete_dynamic_sampler *sampler, l
     // to compute stats. Otherwise, we'd readjust on the next event that comes and thus be operating
     // with very incomplete information
     .last_readjust_time_ns = now_ns,
+    .error = now_ns > 0 ? NULL : ERR_CLOCK_FAIL,
```

Comment:
Wait... why do we need to check for this again, if we've already checked at the beginning of the function?